xbmc/windowing/egl/EGLNativeTypeIMX.cpp

```diff
@@ -338,7 +365,11 @@ bool CEGLNativeTypeIMX::ModeToResolution(std::string mode, RESOLUTION_INFO *res)
   res->iScreen       = 0;
   res->bFullScreen   = true;
   res->iSubtitles    = (int)(0.965 * res->iHeight);
-  res->fPixelRatio   = 1.0f;
+
+  // recalculated PR relatively to startup desktop resolution will result in 1.0f
+  // for all res with same AR and allows use of others without need to calibrate before
+  // by user.
+  res->fPixelRatio   = (float)m_init.fPixelRatio / (res->iScreenWidth/res->iScreenHeight);
```

Comment:
On a DVI monitor, you might also have a 1280x1024 resolution, while the native ratio for the native res is 1.0 - so when you switch to that none native resolution the pixelratio needs to change.

The only problem that I see here is, that we assume that "booting res" == native res or at least has a native ratio. A solution would be parsing EDID to find the width / height in cm of the screen as @mk01 is currently implementing here: https://github.com/mk01/xbmc/commit/9f649819373469c8ed69cbc78f00bec897b4f383